MDPSplitView.m

```diff
@@ -98,13 +102,63 @@ - (void)setValue:(id)value forKey:(NSString *)key
 }
 
 
+#pragma mark - NSView
+
+- (instancetype)initWithFrame:(NSRect)frame
+{
+	self = [super initWithFrame:frame];
+	
+	if (self)
+	{
+		CommonInit(self);
+	}
+	
+	return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)decoder
+{
+	self = [super initWithCoder:decoder];
+	
+	if (self)
+	{
+		CommonInit(self);
+	}
+	
+	return self;
+}
+
++ (id)defaultAnimationForKey:(NSString *)key
+{
+    if ([key hasPrefix:MDPKeyPrefix])
+    {
+        return [CABasicAnimation new];
+    }
+    else
+        return [super defaultAnimationForKey:key];
+}
+
+
 #pragma mark - API
 
 - (void)setPosition:(CGFloat)position ofDividerAtIndex:(NSInteger)dividerIndex animated:(BOOL)animated
 {
     if (animated)
     {
-        [self.animator setValue:@(position) forKey:MDPKeyFromIndex(dividerIndex)];
+		[NSAnimationContext
+			runAnimationGroup:^(NSAnimationContext *context) {
+				@synchronized(self.mdp_animationCounts)
```

Comment:
I don’t understand the need for `@synchronized`—the docs for `+[NSAnimationContext runAnimationGroup:completionHandler:]` don’t mention anything about it occurring in the background, and if `-setPosition:ofDividerAtIndex:animated:` is being called from a background thread I think a) it shouldn’t be and b) we’d want the `self.animator` call synchronized too.